Project/src/main/java/app/RegisterController.java

```diff
@@ -0,0 +1,71 @@
+package app;
+
+import app.models.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.crypto.factory.PasswordEncoderFactories;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+
+@Controller
+public class RegisterController {
+    @Autowired
+    private UserRepository repository;
+
+    private final InMemoryUserDetailsManager inMemoryUserDetailsManager;
+
+    PasswordEncoder encoder =
+            PasswordEncoderFactories.createDelegatingPasswordEncoder();
+    @Autowired
+    public RegisterController(InMemoryUserDetailsManager inMemoryUserDetailsManager) {
+        this.inMemoryUserDetailsManager = inMemoryUserDetailsManager;
+    }
+
+    @GetMapping("/register")
+    public String getRegister(Model model) {
+        model.addAttribute("user", new User());
+        return "register";
+    }
+
+    @PostMapping("/register")
+    public String newBuddy(@ModelAttribute User user, Model model){
```

Comment:
use javax.validation @Valid tags here, and add annotations for constraining name and password requirements. we just need to avoid SQL injection attacks here for now